Add cmake build configuration

p12tic opened the pull request:
cc @simogasp

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,134 @@
+
+cmake_minimum_required(VERSION 3.11)
+
+project(Clp LANGUAGES C CXX)
+
+set(PROJECT_VERSION 1.16.11)
+set(PROJECT_VERSION_MAJOR 1)
+set(PROJECT_VERSION_MINOR 16)
+set(PROJECT_VERSION_PATCH 11)
```
```suggestion
project(Clp LANGUAGES C CXX VERSION 1.16.11)
```

p12tic replied:
Fixed now.

simogasp commented on CMakeLists.txt:
```diff
@@ -0,0 +1,134 @@
+set(INSTALL_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR}/Coin)
+set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+if(MSVC)
```
same comment as coinutils, only if you want

p12tic replied:
Fixed now.

p12tic committed:
Add cmake configuration from vcpkg

The cmake configuration has been taken from vcpkg commit
207fbcb55cef3544f3314ef783d169419f251b3f.

p12tic committed:
Fix paths in CMake configuration

p12tic committed:
Don't define COIN_HAS_CLP in CMake

This macro in hardcoded as defined in the only source file it's used.

p12tic committed:
Use standard install paths for CMake files

p12tic committed:
Cleanup project definition for CMake

p12tic committed:
Don't use GLOB to get list of files to compile in CMake

p12tic committed:
Clean up compile options setup in CMake

p12tic commented:
@simogasp I've addressed your comments.
